frontend/src/components/profile/profileRow.js

```diff
@@ -0,0 +1,72 @@
+import React from "react";
+import {Button, TableCell, TableRow, TextField, Typography} from "@material-ui/core";
+
+export default class ProfileRow extends React.Component {
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            isEditMode: false,
+            value: this.props.value
+        };
+
+        this.handleEdit = this.handleEdit.bind(this);
+        this.handleSave = this.handleSave.bind(this);
+        this.handleValueChange = this.handleValueChange.bind(this);
+    }
+
+    handleEdit() {
+        this.setState({
+            isEditMode: true
+        });
+    }
+
+    handleSave() {
+        // TODO: Upload the new data towards the backend here.
+
+        this.setState({
+            isEditMode: false
+        });
+    }
+
+    handleValueChange(event) {
+        this.setState({value: event.target.value})
+    }
```

Comment:
Did you confirm this works? I have a feeling when the value is changed and the state is updated the `isEditMode` state is also dropped as the whole state object is replaced.

The pattern I usually us is to spread the state and then append new values.
```javascript
handleValueChange(event) {
    this.setState({...this.state, value: event.target.value})
}
```

Reply:
Yes it does work, as the way setState works is that the [states are merged](https://reactjs.org/docs/state-and-lifecycle.html#state-updates-are-merged) not dropped.

Reply:
My bad, its an issue for nested objects since its a shallow merge but not an issue here